Source/CarthageKit/Dependency.swift

```diff
@@ -100,12 +109,29 @@ extension Dependency: Hashable {
 		case let .git(url):
 			return url.hashValue
 
-		case let .binary(url):
-			return url.hashValue
+		case let .binary(binary):
+			return binary.hashValue
 		}
 	}
 }
 
+extension Dependency.Binary: Comparable {
+	public static func == (_ lhs: Dependency.Binary, _ rhs: Dependency.Binary) -> Bool {
+		return lhs.absoluteURL == rhs.absoluteURL &&
+				lhs.representation == rhs.representation
+	}
+
+	public static func < (_ lhs: Dependency.Binary, _ rhs: Dependency.Binary) -> Bool {
+		return lhs.representation.caseInsensitiveCompare(rhs.representation) == .orderedAscending
+	}
+}
+
+extension Dependency.Binary: Hashable {
+	public var hashValue: Int {
+		return absoluteURL.hashValue ^ representation.hashValue
```

Comment:
Not sure if this needs to hash both absoluteURL and representation if the representation is considered a unique id?